llvm-external-projects/iree-compiler-api/python/iree/compiler/tools/binaries.py

```diff
@@ -0,0 +1,283 @@
+# Lint-as: python3
+"""Utilities for locating and invoking compiler tool binaries."""
+
+# Copyright 2020 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import importlib
+import io
+import os
+import platform
+import subprocess
+import sys
+import textwrap
+import threading
+
+from typing import List, Optional, Union
+
+__all__ = [
+    "find_tool",
+    "invoke_immediate",
+    "invoke_pipeline",
+    "get_tool_path",
+    "CompilerToolError",
+]
+
+_BUILTIN_TOOLS = [
+    "ireec",
+]
+
+# In normal distribution circumstances, each named tool is associated with
+# a python module that provides a `get_tool` function for getting its absolute
+# path. This dictionary maps the tool name to the module.
+_TOOL_MODULE_MAP = {
+    "iree-import-tflite": "iree.tools.tflite",
+    "iree-import-xla": "iree.tools.xla",
+    "iree-import-tf": "iree.tools.tf",
+}
+
+# Map of tool module to package name as distributed to archives (used for
+# error messages).
+_TOOL_MODULE_PACKAGES = {
+    "iree.tools.tf": "iree-tools-tf",
+    "iree.tools.tflite": "iree-tools-tflite",
+    "iree.tools.xla": "iree-tools-xla",
+}
```

Comment:
should `iree.tools.core` be listed here with `iree-translate` too?

Reply:
No - that isn't a thing anymore. This will supersede the iree.tools.core package, and now `ireec` is bundled directly with iree.compiler.